apps/hpc-ftsadmin/src/app/utils/codecs.ts

```diff
@@ -0,0 +1,83 @@
+import * as t from 'io-ts';
+import { util } from '@unocha/hpc-data';
+import {
+  DEFAULT_FLOW_TABLE_HEADERS,
+  DEFAULT_KEYWORD_TABLE_HEADERS,
+  DEFAULT_ORGANIZATION_TABLE_HEADERS,
+  type FlowHeaderID,
+  type KeywordHeaderID,
+  type OrganizationHeaderID,
+  type TableHeadersProps,
+} from './table-headers';
+
+const ROWS_PER_PAGE_OPTIONS = [10, 25, 50, 100] as const;
+const ROWS_PER_PAGE = new t.Type<number, number>(
+  'ROWS_PER_PAGE',
+  t.number.is,
+  (v, c) => {
+    if (typeof v === 'number') {
+      return Number.isInteger(v) &&
+        ROWS_PER_PAGE_OPTIONS.some((row) => row === v)
+        ? t.success(v)
+        : t.failure(v, c);
+    } else if (typeof v === 'string') {
+      return /^[0-9]+$/.test(v) &&
```

Comment:
You can substitute `[0-9]` for `\d`
```suggestion
      return /^\d+$/.test(v) &&
```